### PR TITLE
Add doc and schema for documentation field

### PR DESCRIPTION
**Pull request by roadlittledawn**
## Summary

Updates doc and JSON schema for forthcoming pack `documentation` field.

## Notes
* Held off on URL validation for now as there isn't a great, built-in way to do that in JSON schema. [Regex has it's pitfalls](https://json-schema.org/understanding-json-schema/reference/string.html#index-12) and [this](https://www.stephenlewis.me/notes/json-schema-url-validation/) is closest way I found, which felt like it could hit some snags.

## Related issues / PRs
https://github.com/newrelic/newrelic-observability-packs/issues/140
open-install-library-service/pull/56

### CODE REVIEW

File: utils/schemas/main_config.json [hunk 1]
```diff
@@ -370,6 +375,77 @@
       "examples": [
         "https://www.newrelic.com"
       ]
+    },
+    "documentation": {
+      "$id": "#/properties/documentation",
+      "type:": "array",
```
Comment (nr-kkenney) [inline utils/schemas/main_config.json:381]: extra colon in the field name.

File: utils/schemas/main_config.json [hunk 2]
```diff
@@ -370,6 +375,77 @@
...
+                "$id": "#/properties/documentation/items/anyOf/0/properties/url",
+                "type": "string",
+                "title": "The url schema",
+                "description": "The URL for the documentation resource.",
+                "default": "",
+                "examples": [
+                  "https://example.com"
+                ]
+              },
+              "description": {
```
Comment (nr-kkenney) [inline utils/schemas/main_config.json:435]: i think john mentioned limiting this field to ~250 characters.

**Review by nr-kkenney — CHANGES_REQUESTED** [review]
left a couple comments. another general thing is that we should update the tests with a few new cases to validate the new fields.

File: utils/schemas/main_config.json (hunk 2 above)
Comment (aswanson-nr) [reply utils/schemas/main_config.json:435]: I think that was summary field, description is going to stay long form

File: utils/schemas/main_config.json (hunk 1 above)
Comment (roadlittledawn) [reply utils/schemas/main_config.json:381]: 🤦  thanks! i'm now going to make an eye appointment ;)

**Review by nr-kkenney — APPROVED** [review]
lgtm

File: utils/schemas/main_config.json [hunk 3]
```diff
@@ -347,29 +299,86 @@
...
+          }
+        ]
+      ],
+      "items": {
+        "$id": "#/properties/documentation/items",
+        "anyOf": [
+          {
+            "$id": "#/properties/documentation/items/anyOf/0",
+            "type": "object",
+            "title": "The first anyOf schema",
```
Comment (aswanson-nr) [inline utils/schemas/main_config.json:342]: I think this might be left over from copy paste!

**Review by aswanson-nr — CHANGES_REQUESTED** [review]
Saw one comment that looks like it needs to be updated, the rest LGTM!

File: utils/schemas/main_config.json (hunk 3 above)
Comment (roadlittledawn) [reply utils/schemas/main_config.json:342]: which part? the `id` is unique. and i just took the generic `title` value like the others

Comment (aswanson-nr) [reply utils/schemas/main_config.json:342]: Gotcha! could you just update the `description`? Maybe something like "a collection of name, url, and optional description for a piece of documentation"

Comment (roadlittledawn) [reply utils/schemas/main_config.json:342]: added that just now

**Pull request merged**